pkg/vf/vm.go

```diff
@@ -103,6 +103,17 @@ func (cfg *VirtualMachineConfiguration) toVz() (*vz.VirtualMachineConfiguration,
 			return nil, err
 		}
 	}
+	if cfg.config.Timesync != nil && cfg.config.Timesync.VsockPort != 0 {
+		// automatically add the vsock device we'll need for communication over VsockPort
+		vsockDev := VirtioVsock{
+			Port:   cfg.config.Timesync.VsockPort,
+			Listen: false,
+		}
```

Comment:
there's also a constructor `VirtioVsockNew`

Reply:
that constructor returns a different type `VMComponent` using that would would require us to cast the value to `VirtioVsock` again?

Reply:
I tried to use it, but it's a bit messy, there are 2 types, `config.VirtioVsock` and `vf.VirtioVsock` which are aliases for the same type, if I use `config.VirtioVsockNew`, then I need to cast the result to `vf.VirtioVsock`. In addition to that, this returns an error I need to check, and this needs a cast from uint32 to uint.
Easier to manually create the `vf.VirtioVsock` we need even if I agree that it would have been nicer to be able to use `VirtioVsockNew` :)

Reply:
> that constructor returns a different type `VMComponent` using that would would require us to cast the value to `VirtioVsock` again?

Only saw your question after adding my reply, but yes, we'd need an additional cast to the proper type